test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -319,3 +329,203 @@
+                                                      (offset->zone "+07:00")]}
+                         :filter      [:= $times.index 1]
+                         :fields      [$times.dt
+                                       [:expression "to-07"]
+                                       [:expression "to-07-to-09"]]})
+                      mt/rows
+                      first))))
+
+        (testing "filter a converted-timezone column"
+          (is (= ["2004-03-19T18:19:09Z"]
+                 (->> (mt/run-mbql-query
+                        times
+                        {:expressions {"converted" [:convert-timezone $times.dt (offset->zone "+09:00")]
+                                       "hour"       [:get-hour [:expression "converted"]]}
+                         :filter      [:between [:expression "hour"] 17 18]
+                         :fields      [[:expression "converted"]]})
+                      mt/rows
+                      first)))
+          (is (= ["2004-03-19T18:19:09Z"]
+                 (->> (mt/run-mbql-query
+                        times
+                        {:expressions {"converted" [:convert-timezone $times.dt (offset->zone "+09:00")]
+                                       "hour"      [:get-hour [:expression "converted"]]}
+                         :filter      [:= [:expression "hour"] 18]
+                         :fields      [[:expression "converted"]]})
+                      mt/rows
+                      first))))
+
+        (testing "nested query should works"
+          (mt/with-temp Card [card
+                              {:dataset_query
+                               (mt/mbql-query
+                                 times
+                                 {:expressions {"to-07"       [:convert-timezone $times.dt (offset->zone "+07:00")]
+                                                "to-07-to-09" [:convert-timezone [:expression "to-07"] (offset->zone "+09:00")
+                                                               (offset->zone "+07:00")]}
+                                  :filter      [:= $times.index 1]
+                                  :fields      [$times.dt
+                                                [:expression "to-07"]
+                                                [:expression "to-07-to-09"]]})}]
+            (testing "mbql query"
+              (is (= [["2004-03-19T09:19:09Z"
+                       "2004-03-19T16:19:09Z"
+                       "2004-03-19T18:19:09Z"]]
+                     (->> (mt/mbql-query nil {:source-table (format "card__%d" (:id card))})
+                          mt/process-query
+                          mt/rows))))
+
+            (testing "native query"
+              (let [card-tag (format "#%d" (:id card))]
+                ;; FIXME: technically these values should have offset timezone(not just all are 'Z')
+                ;; but we haven't figured out a way to pass the convert_timezone metadata if you use a native query.
```

Comment:
I would remove/update this comment, converted_timezone metadata is no longer with us